File: array/228.py
class Solution:
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        if len(nums) == 1:
            return ['{}'.format(nums[0])]
        
        ret = []
        i = 0
        while i < len(nums):
            cur = nums[i]
            while i+1 < len(nums) and (nums[i+1] - nums[i]) == 1:
                i += 1
            
            if cur is not nums[i]:
                ret.append('{}->{}'.format(cur, nums[i]))
            else:
                ret.append(str(cur))
            i += 1
        return ret
        

File: array/228.cc
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        std::vector<string> ret;
        if (nums.empty())
            return ret;
        for (int i = 0; i < nums.size(); ++i) {
            int cur = nums[i];
            while ((i+1) < nums.size() && (nums[i+1] == (nums[i]+1)))
                ++i;
            if (cur != nums[i]) {
                ret.emplace_back(std::to_string(cur) + "->" + std::to_string(nums[i]));
            } else {
                ret.emplace_back(std::to_string(cur));
            }
        }
        return ret;
    }
};
